### PR TITLE
Feature/nullable reference types

### PR DESCRIPTION
**Pull request by jared-improbable**
(Github wouldn't let me re-open the previous one. :wat:)

### CODE REVIEW

File: GeneratedCode/GeneratedCode.csproj [hunk 1]
```diff
@@ -13,7 +13,7 @@
     <PackageReference Include="Improbable.Stdlib" Version="0.0.4-preview" />
     <PackageReference Include="Improbable.WorkerSdkInterop" Version="14.1.0" />
     <PackageReference Include="Improbable.Schema.Compiler" Version="14.1.0" />
-    <PackageReference Include="Newtonsoft.Json" Version="12.0.3" />
+    <PackageReference Include="Newtonsoft.Json" Version="12.0.2" />
```
Comment (jamiebrynes7) [inline GeneratedCode/GeneratedCode.csproj]: Intended?

File: Improbable/Improbable.CSharpCodeGen/Types.cs [hunk 2]
```diff
@@ -373,51 +294,41 @@ public static string FieldToEquals(FieldDefinition field)
...
                 FieldType.Map => "IndexObject",
-                FieldType.Singular => (field.SingularType.Type.ValueTypeSelector switch
-                {
-                    ValueType.Enum => "GetEnum",
-                    ValueType.Primitive => $"Get{field.SingularType.Type.Primitive}",
-                    ValueType.Type => "GetObject",
-                    _ => throw new ArgumentOutOfRangeException()
-                }),
+                
+                _ when field.HasPrimitive() => $"Get{field.InnerPrimitiveType()}",
```
Comment (zeroZshadow) [inline Improbable/Improbable.CSharpCodeGen/Types.cs]: I'm curious what kind of IL this generates

**Review by zeroZshadow — APPROVED** [review]

**Comment by zeroZshadow** [comment]
LGTM, though its pretty large for a PR

**Review by jamiebrynes7 — APPROVED** [review]
LGTM

File: GeneratedCode/GeneratedCode.csproj (hunk 1 above)
Comment (jared-improbable) [reply GeneratedCode/GeneratedCode.csproj]: Nope, bad merge. I haven't pushed the fix up yet

File: Improbable/Directory.Build.targets [hunk 3]
```diff
@@ -0,0 +1,12 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (jared-improbable) [inline Improbable/Directory.Build.targets:1]: This is a workflow improvement for local iteration that I accidentally checked in

Comment (jared-improbable) [reply Improbable/Directory.Build.targets:1]: (still WIP on the consumer side of things), this part is complete

**Pull request merged**